src/cloudsc2_tl_loki/error.F90

```diff
@@ -2,16 +2,22 @@ MODULE ERROR_MOD
   USE PARKIND1, ONLY: JPIM, JPRB
   IMPLICIT NONE
   CONTAINS
-  SUBROUTINE ERROR_NORM(NLON, FIELD, PERT5, PERT, ZNORM, ZCOUNT, ZLAMBDA)
+  SUBROUTINE ERROR_NORM(KIDIA,KFDIA,NLON, FIELD, PERT5, PERT, ZNORM, ZCOUNT, ZLAMBDA)
     INTEGER(KIND=JPIM), INTENT(IN) :: NLON
+    INTEGER(KIND=JPIM),INTENT(IN)    :: KIDIA
+    INTEGER(KIND=JPIM),INTENT(IN)    :: KFDIA
     REAL(KIND=JPRB), INTENT(IN) :: FIELD(:,:), PERT5(:,:), PERT(:,:)
     REAL(KIND=JPRB), INTENT(IN) :: ZLAMBDA
     REAL(KIND=JPRB), INTENT(INOUT) :: ZNORM, ZCOUNT
-
-    IF (ABS(SUM(PERT(1:NLON,:)*ZLAMBDA)) > EPSILON(ZLAMBDA) ) THEN
-      ZCOUNT = ZCOUNT + 1._JPRB
-      ZNORM = ZNORM + ABS(SUM(FIELD(1:NLON,:)-PERT5(1:NLON,:)) / SUM(PERT(1:NLON,:)*ZLAMBDA))
-    END IF
+    INTEGER(KIND=JPIM) :: JK, JL
+    DO JL=KIDIA, KFDIA
+      IF(JL == KIDIA) THEN
```

Comment:
Note to self: Why is this used here? This completely goes against the idea of applying this norm over the block, no?